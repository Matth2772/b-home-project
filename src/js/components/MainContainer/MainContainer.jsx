import React, { Component } from 'react';
import Container from 'react-bootstrap/Container';
import SearchField from '../SearchField/SearchField';
import Results from '../Results/Results';
import HackerNewsService from '../../services/HackerNews.service';

class MainContainer extends Component {
  constructor() {
    super();
    this.state = {
      query: '',
      results: [],
      requestsOut: 0,
    };

    this.search = this.search.bind(this);
  }

  search(q) {
    this.setState(prevState => ({
      requestsOut: prevState.requestsOut + 1,
      query: q,
    }));

    HackerNewsService.getSearch(q)
      .then((data) => {
        this.setState(prevState => ({
          requestsOut: prevState.requestsOut - 1,
        }), this.updateResults(q, data.hits));
      })
      .catch(() => {
        this.setState(prevState => ({
          requestsOut: prevState.requestsOut - 1,
        }));
      });
  }

  updateResults(q, results) {
    return () => {
      const { query, requestsOut } = this.state;
      if (q === query && requestsOut === 0) {
        this.setState({ results });
      }
    };
  }

  render() {
    const { results } = this.state;

    return (
      <Container>
        <SearchField search={this.search} />
        <Results results={results} />
      </Container>
    );
  }
}

export default MainContainer;
