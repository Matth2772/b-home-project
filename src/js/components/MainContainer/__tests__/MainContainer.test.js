/* eslint-disable */
import React from 'react';
import MainContainer from '../MainContainer';
import renderer from 'react-test-renderer';

test('MainContainer rendering well', () => {
  const component = renderer.create(
    <MainContainer/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
