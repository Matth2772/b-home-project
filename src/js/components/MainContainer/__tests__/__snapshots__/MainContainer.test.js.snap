// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="container"
>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="search"
    >
      Hacker News Search
    </label>
    <input
      className="form-control"
      id="search"
      onChange={[Function]}
      placeholder="Search news"
      type="email"
    />
  </div>
  <form
    className=""
  />
</div>
`;

exports[`MainContainer rendering well 1`] = `
<div
  className="container"
>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="search"
    >
      Hacker News Search
    </label>
    <input
      className="form-control"
      id="search"
      onChange={[Function]}
      placeholder="Search news"
      type="email"
    />
  </div>
  <form
    className=""
  />
</div>
`;
